File: LevelOrderTraversal.cpp
vector<vector<int>> levelOrder(Node *root) {
        // Your code here
        queue<Node*> queue;
    queue.push(root);
    vector<vector<int>> res;
    
    while (!queue.empty()) {
        int sz = queue.size();
        vector<int> temp;
        
        for (int i = 0; i < sz; ++i) {
            Node* curr = queue.front();
            queue.pop();
            temp.push_back(curr->data);
            
            if (curr->left) {
                queue.push(curr->left);
            }
            
            if (curr->right) {
                queue.push(curr->right);
            }
        }
        res.push_back(temp);
    }
    
    return res;
    }

File: LevelOrderTraversal.java
public ArrayList<ArrayList<Integer>> levelOrder(Node root) {
        // Your code here
        
        Queue<Node> q = new LinkedList<>();
        q.add(root);
        ArrayList<ArrayList<Integer>> res = new ArrayList<>();
        while(!q.isEmpty()){
            int sz = q.size();
            ArrayList<Integer> levelElements = new ArrayList<>();
            for(int i = 0; i < sz; ++i){
                Node curr = q.poll();
                levelElements.add(curr.data);
                if(curr.left != null){
                    q.add(curr.left);
                }
                
                if(curr.right != null){
                    q.add(curr.right);
                }
            }
            res.add(levelElements);
            
        }
        return res;
    }
